qiita_pet/handlers/artifact_handlers/base_handlers.py

```diff
@@ -370,6 +371,10 @@ def artifact_patch_request(user, artifact_id, req_op, req_path, req_value=None,
                            "directly to <a href='mailto:{0}'>{0}</a>.".format(
                                email_to))
                     raise QiitaHTTPError(400, msg)
+            else:
+                msg = '%s changed artifact %s (study %d) to %s' % (
+                    user.email, artifact_id, sid, req_value)
+                LogEntry.create('Warning', msg)
```

Comment:
Would it be better for the `LogEntry.create` to occur within the `property.set` on `artifact.visibility`? That way, and if I understand this correctly, a log entry would be recorded for any visibility change whereas I think the log entry here would only be recorded if done via the GUI?

Reply:
Yes, good question and possible but that will require that [the method](https://github.com/qiita-spots/qiita/blob/master/qiita_db/artifact.py#L754) accept the user as a parameter - which might complicate things a bunch.

Reply:
Ya... was worried this would be a complex item, which I think either would require changing that from a property to a method, or allowing the `Artifact` constructor to accept the user so the object knew about about them at the time of the check. Either would be complex and definitely out of scope here.